File: pantry-tracker-backend/database_operations/getIngredientList.py
from flask import Blueprint, jsonify
from sqlalchemy import text
from extensions import db

getIngredientList_bp = Blueprint('getIngredientList', __name__)
@getIngredientList_bp.route('/get-ingredient-list', methods=['GET'])

def get_ingredient_list():
    try:
        result = db.session.execute(text('SELECT mp.aliased_ingredient_name, mp.quantity, u.unit_name FROM my_pantry mp JOIN units u ON mp.unit_id = u.id WHERE mp.quantity > 0 ORDER BY mp.aliased_ingredient_name'))
        ingredients = [{'name': row[0], 'quantity': row[1], 'unitID': row[2]} for row in result]
        return jsonify({'ingredients': ingredients})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

File: pantry-tracker-frontend/src/supabase_db_operations/getIngredientList.js
import supabaseClient from '../utils/SupabaseClient';

const getIngredientList = async (userID) => {
    try {
        const { data: userIngredients, error } = await supabaseClient
            .from('user_ingredients')
            .select(`
                aliased_ingredient_name,
                p_quantity,
                units:p_unit_id(unit_name)
            `)
            .eq('user_id', userID)
            .gt('p_quantity', 0)
            .order('aliased_ingredient_name', { ascending: true});

        if (error) {
            throw (error);
        };

        const ingredients = userIngredients.map(ingredient => ({
            name: ingredient.aliased_ingredient_name,
            quantity: ingredient.p_quantity,
            unitID: ingredient.units.unit_name
        }));
        
        return ingredients;

    }
    catch (error) {
        console.error(error);
        return { 'error': error.message };
    };
};

export default getIngredientList;